packages/connect-form/src/hooks/connection-options-handler.ts

```diff
@@ -0,0 +1,175 @@
+import { ConnectionOptions } from "mongodb-data-service";
+import { MARKABLE_FORM_FIELD_NAMES } from "../constants/markable-form-fields";
+import { SSHFormErrors } from "../utils/connect-form-errors";
+import { ConnectFormState } from "./use-connect-form";
+import { checkForInvalidCharacterInHost } from "../utils/check-for-invalid-character-in-host";
+
+export type SSHConnectionOptions = NonNullable<ConnectionOptions['sshTunnel']>;
+
+export type SSHType = 'none' | 'password' | 'identity' | 'socks';
+
+export interface UpdateConnectionOptions {
+  type: 'update-connection-options';
+  currentTab: SSHType;
+  key: keyof SSHConnectionOptions;
+  value: string | number;
+}
+
+export function handleUpdateConnectionOptions(action: UpdateConnectionOptions, {
+  errors: initialErrors,
```

Comment:
Why do we need these initial errors? Not sure I understand their use case

Reply:
want to maintain the existing errors in the state, without overwriting them completely with new ssh form errors.